src/GeneIDTypeDetector.ts

```diff
@@ -0,0 +1,32 @@
+function detectIDType(data: any[], accessor: (row: any) => string, sampleSize: number) : Promise<number> {
+  return new Promise((resolve) => {
```

Comment:
why wrapping in a new Promise construct, when Promise.resolve would also do the job?